includes/class-comment-walker.php

```diff
@@ -0,0 +1,173 @@
+					$callback = $args['callback']['all'];
+				}
+			}
+			if ( $callback ) {
+				ob_start();
+				call_user_func( $callback, $comment, $args, $depth );
+				$output .= ob_get_clean();
+				return;
+			}
+		}
+
+		/* Implements an argument that will just output an avatar */
+		if ( ! isset( $args['avatar_only'] ) ) {
+			$args['avatar_only'] = true;
+		}
+
+		/* Implements an argument that optionally overlays an icon on top of the profile image, applies only to the avatar only output  */
+		if ( ! isset( $args['overlay'] ) ) {
+			$args['overlay'] = true;
+		}
+
+		if ( 'comment' === $comment->comment_type ) {
+			add_filter( 'comment_text', array( $this, 'filter_comment_text' ), 40, 2 );
+		}
+
+		// Maintain the original pingback and trackback output.
+		if ( ( 'pingback' === $comment->comment_type || 'trackback' === $comment->comment_type ) && $args['short_ping'] ) {
+			ob_start();
+			$this->ping( $comment, $depth, $args );
+			$output .= ob_get_clean();
+		} elseif ( $args['avatar_only'] ) {
+			ob_start();
+			$this->avatar_only( $comment, $depth, $args );
+			$output .= ob_get_clean();
+
+		} elseif ( 'html5' === $args['format'] ) {
+			ob_start();
+			$this->html5_comment( $comment, $depth, $args );
+			$output .= ob_get_clean();
+		} else {
+			ob_start();
+			$this->comment( $comment, $depth, $args );
+			$output .= ob_get_clean();
+		}
+
+		if ( 'comment' === $comment->comment_type ) {
+			remove_filter( 'comment_text', array( $this, 'filter_comment_text' ), 40 );
+		}
+	}
+
+	/**
+	 * Ends the element output, if needed.
+	 *
+	 * @see Walker::end_el()
+	 * @see wp_list_comments()
+	 *
+	 * @param string     $output      Used to append additional content. Passed by reference.
+	 * @param WP_Comment $data_object Comment data object.
+	 * @param int        $depth       Optional. Depth of the current comment. Default 0.
+	 * @param array      $args        Optional. An array of arguments. Default empty array.
+	 */
+	public function end_el( &$output, $data_object, $depth = 0, $args = array() ) {
+		if ( ! empty( $args['end-callback'] ) ) {
+			ob_start();
+			call_user_func(
+				$args['end-callback'],
+				$data_object, // The current comment object.
+				$args,
+				$depth
+			);
+			$output .= ob_get_clean();
+			return;
+		}
+		if ( 'div' === $args['style'] ) {
+			$output .= "</div><!-- #comment-## -->\n";
+		} else {
+			$output .= "</li><!-- #comment-## -->\n";
+		}
+	}
+
+	/**
+	 * Outputs a comment as just a profile picture.
+	 *
+	 * @param WP_Comment $comment The comment object.
+	 * @param int        $depth   Depth of the current comment.
+	 * @param array      $args    An array of arguments.
+	*/
+	protected function avatar_only( $comment, $depth, $args ) {
+		$tag = ( 'div' === $args['style'] ) ? 'div' : 'li';
+
+		$title = get_comment_text( $comment, $args );
+		// Optionally overlay an icon.
+		$overlay = '';
+		?>
+		<<?php echo $tag; ?> id="comment-<?php comment_ID(); ?>" <?php comment_class( '', $comment ); ?>>
```

Comment:
use a template for this?

Reply:
The new code allows someone to write callbacks, the way comments are currently done, per comment type. The goal was to try and make it as close to the stock method as possible. I can move this to a template, but it shouldn't be able to be overwritten here as it is further upstream.